utils/api/logs.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers } from 'ethers'
+import { LogDescription } from 'ethers/lib/utils'
+import { API_ENDPOINT, pretreat } from './utils'
+
+interface Log {
+  attributes: Attributes
+  id: number
+  relationships: any
+  type: string
+}
+
+interface Attributes {
+  abi: Fragment[]
+  address_hash: string
+  block_number: number
+  data: string
+  first_topic: string
+  second_topic: string
+  third_topic: string
+  fourth_topic: string
+  transaction_hash: string
+}
+
+interface Input {
+  indexed?: boolean
+  internalType: string
+  name: string
+  type: string
+}
+
+interface Output {
+  internalType: string
+  name: string
+  type: string
+}
+
+interface Meta {
+  current_page?: number
+  total_page?: number
+}
+
+interface Fragment {
+  anonymous?: boolean
+  inputs: Input[]
+  name: string
+  type: 'event' | 'function' | string
+  outputs?: Output[]
+  stateMutability?: string
+}
+
+interface Raw {
+  data: Log[]
+  included: any[]
+  meta: Meta
+}
+
+export interface ParsedEventLog {
+  id: number
+  parsedLog: LogDescription
+  addressHash: string
+  txHash: string
+  data: string
+  blockNumber: number
+  topics: string[]
+}
+
+export const getEventLogsListRes = (raw: Raw): ParsedEventLog[] => {
+  return raw.data.map(({ attributes, id }) => {
+    const log3 = [attributes.first_topic, attributes.second_topic, attributes.third_topic]
+    const log4 = [...log3, attributes.fourth_topic]
```

Comment:
It's not the correct way to handle `null` topic since topics[0...3] could be `null` too.
Use `0x` if a topic is a `null`